### PR TITLE
Cover Webservice key management with Behat tests

### PR DESCRIPTION
**Pull request by sarjon**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Adds Behat tests for Webservice key management
| Type?         | improvement
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Build should be green.

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/14109)

### CODE REVIEW

File: tests/Integration/Behaviour/Features/Context/SharedStorage.php [hunk 1]
```diff
@@ -77,4 +77,14 @@ public function set($key, $resource)
     {
         $this->storage[$key] = $resource;
     }
+
+    /**
+     * @param string $key
+     */
+    public function clear($key)
+    {
+        if (isset($this->storage[$key])) {
```
Comment (PierreRambaud) [inline tests/Integration/Behaviour/Features/Context/SharedStorage.php:86]: array_key_exists instead? Maybe we'll want to keep null values. Wdyt?

Comment (sarjon) [reply tests/Integration/Behaviour/Features/Context/SharedStorage.php:86]: Hmm, not sure if it's worth covering case that does not exist and. :thinking: @matks wdyt?

**Comment by sarjon** [comment]
"Segmentation fault (core dumped)" happens too often recently during build. :/

**Comment by matks** [comment]
> "Segmentation fault (core dumped)" happens too often recently during build. :/

Yep, I asked Travis about it https://twitter.com/mathieuKs/status/1134444500523393024

File: tests/Integration/Behaviour/Features/Context/SharedStorage.php (hunk 1 above)
Comment (matks) [reply tests/Integration/Behaviour/Features/Context/SharedStorage.php:86]: Errr ... yeah @PierreRambaud the storage is intended to keep track of items created in our behat tests. Why would be keep a `null` value ? 🤔

**Comment by sarjon** [comment]
This one is ready for review.

File: tests/Integration/Behaviour/Features/Context/Domain/WebserviceKeyFeatureContext.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        $data['is_enabled'] = (bool) $data['is_enabled'];
+        $data['shop_association'] = [
+            SharedStorage::getStorage()->get($data['shop_association'])->id,
+        ];
+
+        SharedStorage::getStorage()->set(sprintf('%s_properties', $reference), $data);
+    }
+
+    /**
+     * @Then /^I specify "(View|Add|Modify|Delete|Fast view)" permission for "([^"]*)" resources for new webservice key "(.*)"$/
```
Comment (matks) [inline tests/Integration/Behaviour/Features/Context/Domain/WebserviceKeyFeatureContext.php:56]: 👍

**Review by matks — APPROVED** [review]

**Comment by sarjon** [comment]
:shipit: ?

File: tests/Integration/Behaviour/Features/Context/SharedStorage.php (hunk 1 above)
Comment (PierreRambaud) [reply tests/Integration/Behaviour/Features/Context/SharedStorage.php:86]: Because you wanted to tried with an empty data :p

**Comment by matks** [comment]
> :shipit: ?

🚀

**Comment by PierreRambaud** [comment]
> > :shipit: ?
> 
> 🚀 

:taco:

**Pull request merged**